web/modules/custom/custom_page/custom_page.services.yml

```diff
@@ -4,3 +4,7 @@ services:
     arguments: ['@entity.manager']
     tags:
      - { name: event_subscriber }
+  custom_page.og_menu_link_updater:
+    class: Drupal\custom_page\CustomPageOgMenuLinksUpdater
+    arguments: ['@entity_type.manager', '@plugin.manager.menu.link']
+    lazy: true
```

Comment:
Services aren't lazy by default? Then we probably need to apply this on many different places.

Reply:
No, they are instantiated on each request. As a general rule, every service that doesn't use a factory class is eligible to be made lazy. Unless we know that we need it in each request. I was about to open a ticket for that.

Reply:
Opened https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-4047 to make several services lazy.

Reply:
@brummbar explained this to me. By default services themselves are actually lazy loaded (instantiated on demand by the service container), but the dependencies of the service are instantiated directly, even when they may not be needed by all methods on the service. So marking them as lazy and providing the proxy class allows to lazy load _the dependencies of the service_ on demand. This sounds really interesting for many of the services we have in Joinup, especially the helper services that have a lot of dependencies.

Reply:
That's correct. More details here https://www.drupal.org/node/1973618